core/src/dird/ua_prune.cc

```diff
@@ -230,6 +231,53 @@ bool PruneCmd(UaContext* ua, const char* cmd)
       }
 
       return PruneDirectory(ua, client);
+
+    case 5: /* prune all */
+    {
+      if (FindArg(ua, "volumes")) {
+        if ((FindArgWithValue(ua, NT_("pool")) >= 0)
+            || ua->AclHasRestrictions(Pool_ACL)) {
+          pool = get_pool_resource(ua);
+        } else {
+          pool = nullptr;
+        }
+        bool result = true;
+        db_list_ctx volumenames;
+        ua->db->GetAllVolumeNames(&volumenames);
+        for (const auto& volume : volumenames) {
+          if (!SelectPoolAndMediaDbr(ua, &pr, &mr, volume.c_str())) {
+            return false;
+          }
+          if (pool) {
+            if (strcmp(pool->resource_name_, pr.Name)) { continue; }
+          }
+
+          if (mr.Enabled == VOL_ARCHIVED) {
+            ua->ErrorMsg(
+                _("Cannot prune Volume \"%s\" because it is archived.\n"),
+                mr.VolumeName);
+            result = false;
+            continue;
+          }
+
+          std::string msg("Volume");
+          msg.append(" (" + volume + ")");
+          if (pool && pool->VolRetention > 0) {
+            if (!ConfirmRetention(ua, &pool->VolRetention, msg.c_str())) {
```

Comment:
`pool->VolRetention` may differ from `mr.VolRetention`. In that case `mr.VolRetention` takes precedence.
Thus, checking only `mr.VolRetention` is probably the right thing to do.

Reply:
makes sense